src/groups/mqb/mqbauthn/mqbauthn_basicauthenticator.cpp

```diff
@@ -0,0 +1,212 @@
+// Copyright 2016-2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mqbauthn_basicauthenticator.h>
+
+// MQB
+#include <mqbcfg_messages.h>
+#include <mqbplug_authenticator.h>
+
+// BDE
+#include <bdlt_timeunitratio.h>
+#include <bsl_iostream.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bsla_annotations.h>
+#include <bslma_allocator.h>
+#include <bslma_managedptr.h>
+#include <bsls_assert.h>
+
+namespace BloombergLP {
+namespace mqbauthn {
+
+namespace {
+
+const int k_TIMEOUT_SECONDS = 600;
+
+}  // namespace
+
+// -------------------------------
+// class BasicAuthenticationResult
+// -------------------------------
+
+BasicAuthenticationResult::BasicAuthenticationResult(
+    bsl::string_view   principal,
+    bsls::Types::Int64 lifetimeMs,
+    bslma::Allocator*  allocator)
+: d_principal(principal, allocator)
+, d_lifetimeMs(lifetimeMs)
+{
+}
+
+BasicAuthenticationResult::~BasicAuthenticationResult()
+{
+}
+
+bsl::string_view BasicAuthenticationResult::principal() const
+{
+    return d_principal;
+}
+
+const bsl::optional<bsls::Types::Int64>&
+BasicAuthenticationResult::lifetimeMs() const
+{
+    return d_lifetimeMs;
+}
+
+// ------------------------
+// class BasicAuthenticator
+// ------------------------
+
+BasicAuthenticator::BasicAuthenticator(
+    const mqbcfg::AuthenticatorPluginConfig* config,
+    bslma::Allocator*                        allocator)
+: d_allocator_p(allocator)
+, d_authenticatorConfig_p(config)
+, d_credentials(allocator)
+, d_isStarted(false)
+{
+    if (!config) {
+        return;
+    }
+
+    // Load the configured key-values as username and password
+    bsl::vector<mqbcfg::PluginConfigKeyValue>::const_iterator it =
+        config->configs().cbegin();
+    for (; it != config->configs().cend(); ++it) {
+        if (it->value().isStringValValue()) {
+            d_credentials[it->key()] = it->value().stringVal();
```

Comment:
Same with Anon Pass, can print something if there is a type mismatch or if there are no values(?)